### DISCUSSION TITLE
How Can I ingest multiple prometheus data and create visualization on grafana?

### DISCUSSION THREAD

**Discussion by ankitola** [grafana/mimir#1656] [Help and support]

**Comment by pstibrany** [comment]
Hello, to ingest data from multiple Prometheus servers, you need to configure each Prometheus server to `remote_write` data to Mimir.

See this section of "Getting started with Grafana Mimir" tutorial: https://grafana.com/docs/mimir/latest/operators-guide/getting-started/#configure-prometheus-to-write-to-grafana-mimir

**Reply by bulmust** [reply]
<https://grafana.com/docs/mimir/latest/get-started/#configure-prometheus-to-write-to-grafana-mimir>

**Comment by ankitola** [comment]
How I will differentiate the data on the Grafana dashboard for multiple  Prometheus?
Like- If I am having one Prometheus configured on Kubernetes and another configured on my VM so how will I check the data separately for these two Prometheus on the dashboard.
Any of the tag should I use there on every Prometheus like namespace or cluster?
Also, will this work for existing Prometheus which are running without containers?

**Comment by pstibrany** [comment]
> How I will differentiate the data on the Grafana dashboard for multiple Prometheus?

Typically you don't need to differentiate data based on Prometheus servers. They will simply forward metrics for jobs they know about, and individual metrics will have correct `job` and `instance` labels.

If you however DO need that, you can configure `external_labels` in the `global` section of the Prometheus. [Reference](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#configuration-file). If for example your Prometheus in Kubernetes has following external_labels:

```yaml
global:
  external_labels:
    prometheus: kubernetes
```

And Prometheus running in VM has:

```yaml
global:
  external_labels:
    prometheus: vm
```

then all data sent by these Prometheus servers will have either `prometheus="kubernetes"` or `prometheus="vm"` labels, depending on which Prometheus sent the data.

> Also, will this work for existing Prometheus which are running without containers?

Any Prometheus regardless of how it was started can be configured to remote-write to Mimir. (Not sure if that was your question)
